src/metadata.mjs

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+/**
+ * This method allows us to handle creation of Metadata entries
+ * within the current scope of API docs being parsed
+ *
+ * This can be used disconnected with a specific file, and can be aggregated
+ * to many files to create a full Navigation for a given version of the API
+ *
+ * @param {InstanceType<typeof import('github-slugger').default>} slugger A GitHub Slugger
+ * @param {ReturnType<typeof remark>} remarkProcessor A Remark processor
```

Comment:
`ReturnType<typeof remark>` seems to be any, it might be better to use it like this;

```suggestion
 * @param {import('unified').Processor} remarkProcessor A Remark processor
```